app/models/form/sales/questions/prevloc.rb

```diff
@@ -14,390 +13,396 @@ def initialize(id, hsh, page)
-    "E09000028" => "Southwark",
-    "E07000213" => "Spelthorne",
-    "E07000240" => "St Albans",
-    "E08000013" => "St. Helens",
-    "E07000197" => "Stafford",
-    "E07000198" => "Staffordshire Moorlands",
-    "E07000243" => "Stevenage",
-    "S12000030" => "Stirling",
-    "E08000007" => "Stockport",
-    "E06000004" => "Stockton-on-Tees",
-    "E06000021" => "Stoke-on-Trent",
-    "E07000221" => "Stratford-on-Avon",
-    "E07000082" => "Stroud",
-    "E08000024" => "Sunderland",
-    "E07000214" => "Surrey Heath",
-    "E09000029" => "Sutton",
-    "E07000113" => "Swale",
-    "W06000011" => "Swansea",
-    "E06000030" => "Swindon",
-    "E08000008" => "Tameside",
-    "E07000199" => "Tamworth",
-    "E07000215" => "Tandridge",
-    "E07000045" => "Teignbridge",
-    "E06000020" => "Telford and Wrekin",
-    "E07000076" => "Tendring",
-    "E07000093" => "Test Valley",
-    "E07000083" => "Tewkesbury",
-    "E07000114" => "Thanet",
-    "E07000102" => "Three Rivers",
-    "E06000034" => "Thurrock",
-    "E07000115" => "Tonbridge and Malling",
-    "E06000027" => "Torbay",
-    "W06000020" => "Torfaen",
-    "E07000046" => "Torridge",
-    "E09000030" => "Tower Hamlets",
-    "E08000009" => "Trafford",
-    "E07000116" => "Tunbridge Wells",
-    "E07000077" => "Uttlesford",
-    "W06000014" => "Vale of Glamorgan",
-    "E07000180" => "Vale of White Horse",
-    "E08000036" => "Wakefield",
-    "E08000030" => "Walsall",
-    "E09000031" => "Waltham Forest",
-    "E09000032" => "Wandsworth",
-    "E06000007" => "Warrington",
-    "E07000222" => "Warwick",
-    "E07000103" => "Watford",
-    "E07000216" => "Waverley",
-    "E07000065" => "Wealden",
-    "E07000156" => "Wellingborough",
-    "E07000241" => "Welwyn Hatfield",
-    "E06000037" => "West Berkshire",
-    "E07000047" => "West Devon",
-    "S12000039" => "West Dunbartonshire",
-    "E07000127" => "West Lancashire",
-    "E07000142" => "West Lindsey",
-    "S12000040" => "West Lothian",
-    "E07000181" => "West Oxfordshire",
-    "E07000245" => "West Suffolk",
-    "E09000033" => "Westminster",
-    "E08000010" => "Wigan",
-    "E06000054" => "Wiltshire",
-    "E07000094" => "Winchester",
-    "E06000040" => "Windsor and Maidenhead",
-    "E08000015" => "Wirral",
-    "E07000217" => "Woking",
-    "E06000041" => "Wokingham",
-    "E08000031" => "Wolverhampton",
-    "E07000237" => "Worcester",
-    "E07000229" => "Worthing",
-    "W06000006" => "Wrexham",
-    "E07000238" => "Wychavon",
-    "E07000128" => "Wyre",
-    "E07000239" => "Wyre Forest",
-    "E06000014" => "York",
-    "N92000002" => "Northern Ireland",
-    "S92000003" => "Scotland",
-    "W92000004" => "Wales",
-    "9300000XX" => "Outside UK",
-  }.freeze
+  def answer_options
+    {
+      "" => "Select an option",
+      "S12000033" => "Aberdeen City",
+      "S12000034" => "Aberdeenshire",
+      "E07000223" => "Adur",
+      "E07000026" => ("Allerdale" if form.start_date.year < 2023),
```

Comment:
How would you do this if we didn't have local authorities statically defined, and they were in the database instead?

Reply:
I guess similar to what I've done in locations. Have the list of all and then filter down based on the year. I can change that here too if that seems more appropriate

Reply:
What do you think about something like this? We might use fixed dates instead of years, but the principle remains the same.

```ruby
LocalAuthority.active(2022) 
# or
LocalAuthority.active(Time.today)
```

```diff
diff --git a/app/models/local_authority.rb b/app/models/local_authority.rb
new file mode 100644
index 00000000..0014f357
--- /dev/null
+++ b/app/models/local_authority.rb
@@ -0,0 +1,3 @@
+class LocalAuthority < ApplicationRecord
+  scope :active, ->(year) { where("? BETWEEN start_year AND end_year", year) }
+end
diff --git a/db/migrate/20230308091405_create_local_authorities.rb b/db/migrate/20230308091405_create_local_authorities.rb
new file mode 100644
index 00000000..3581ce7b
--- /dev/null
+++ b/db/migrate/20230308091405_create_local_authorities.rb
@@ -0,0 +1,12 @@
+class CreateLocalAuthorities < ActiveRecord::Migration[7.0]
+  def change
+    create_table :local_authorities do |t|
+      t.string :code
+      t.string :name
+      t.integer :start_year
+      t.integer :end_year
+
+      t.timestamps
+    end
+  end
+end
```